eng/scripts/Track1CodeCheck.ps1

```diff
@@ -0,0 +1,236 @@
+    )
+    $obj | Get-Member -MemberType NoteProperty | ForEach-Object {
+        $key = $_.Name
+        [PSCustomObject]@{Key = $key; Value = $obj."$key" }
+    }
+}
+
+try {
+    # Get RP Mapping
+    $RPMapping = [ordered]@{ }
+    $readmePath = ''
+    git clone https://github.com/Azure/azure-rest-api-specs.git ../azure-rest-api-specs
+    $folderNames = Get-ChildItem ../azure-rest-api-specs/specification
+    $folderNames | ForEach-Object {
+        $folderName = ''
+        $readmePath = "../azure-rest-api-specs/specification/$($_.Name)/resource-manager/readme.csharp.md"
+        if (Test-Path $readmePath) {
+            $folderName = Find-Mapping $readmePath
+        }
+        if (($folderName -eq '') -or ($folderName -match "\$")) {
+            $readmePath = "../azure-rest-api-specs/specification/$($_.Name)/resource-manager/readme.md"
+            if (Test-Path $readmePath) {
+                $folderName = Find-Mapping $readmePath
+            }
+        }
+        if (($folderName -notmatch "\$") -and (!$RPMapping.Contains($folderName)) -and ($folderName -ne '')) {
+            $RPMapping += @{ $folderName = "$($_.Name)" }
+        }
+    }
+
+    # Get Metadata file path
+    $Response = Invoke-WebRequest -URI https://api.github.com/repos/$Env:REPOSITORY_NAME/pulls/$Env:PULLREQUEST_ID/files
+    $changeList = $Response.Content | ConvertFrom-Json
+    if ($Response.RelationLink) {
+        $lastLink = $Response.RelationLink.Get_Item('last')
+        $lastPage = $lastLink.Substring($lastLink.indexof("=") + 1)
+        for ($i = 2; $i -le $lastPage; $i++) {
+            $Response = Invoke-WebRequest -URI https://api.github.com/repos/$Env:REPOSITORY_NAME/pulls/$Env:PULLREQUEST_ID/files?page=$i
+            $changeList += $Response.Content | ConvertFrom-Json
+        }
+    }
+    $mataPath = @()
+    $rpIndex = @()
+    $folderName = @()
+    $changeList | ForEach-Object {
+        $fileName = $_.filename
+        if ($fileName -match 'eng/mgmt/mgmtmetadata') {
+            $mataPath += $fileName
+        }
+    }
+    $changeList | ForEach-Object {
+        $fileName = $_.filename
+        if ($fileName -match 'sdk/') {
+            $name = $fileName.substring(4, (($fileName.indexof('/Microsoft') - 4)))
+            If ($folderName -notcontains $name) {
+                $folderName += $name
+            }
+        }
+    }
+    foreach ($item in $folderName) {
+        $rpName = $RPMapping.Get_Item($item)
+        if ($rpName) {
+            If ($rpIndex -notcontains $rpName) {
+                $rpIndex += $rpName
+            }
+        }
+        else {
+            LogError "Can't get proper RP name with folder $item"
+        } 
+    }
+    $rpIndex | ForEach-Object {
+        $path = "eng/mgmt/mgmtmetadata/$_" + "_resource-manager.txt"
+        if ($mataPath -notcontains $path) {
+            $mataPath += $path
+        }
+    }
+
+    # Install AutoRest    
+    Invoke-Block {
+        & npm install -g autorest
```

Comment:
Perhaps a naïve question. Should we be pinning to a specific version of AutoRest here? If the generated files change as a result of the version of AutoRest we use then we would block. That might not be such a problem for master, but if we need to hotfix a prior release for any reason that could complicate the release process.

I'm not sure if this is a concern that we need to deal with though.

Reply:
Thanks for the comment. This is indeed a potential issue which needs to be handled. We'll record and keep tracking it. 
For now, I used autorest version v2 and csharp version 2.3.90, should be stable for a while. 
If there is a special verison used in release PR, pushed to *-preview brach triggers this pipeline, causes CI failures, which are expected and I think we can manually verify and merge it.
In furture, we may setup a configuration file for the pipeline or other approaches to handle it.

Reply:
Did you have a link to the issue?

Reply:
https://github.com/Azure/autorest.csharp/pull/106
If we use autorest by default parameters not a specific version in pipeline, it will use csharp version 2.3.79, not the latest one. Doesn't contains changes I've mentioned above. So we set version to 2.3.90 to avoid that.